Zilon.Core/Zilon.Core.Tests/World/ResourceMaterializationMapTests.cs

```diff
@@ -0,0 +1,241 @@
+                        return sector;
+                    });
+                });
+
+            var sectorGenerator = sectorGeneratorMock.Object;
+            var locationSchemes = CreateBiomSchemes();
+
+            var schemeRoundCounter = 0;
+            var rollerMock = new Mock<IBiomeSchemeRoller>();
+            rollerMock.Setup(x => x.Roll()).Returns(() =>
+            {
+                schemeRoundCounter++;
+                if (schemeRoundCounter >= locationSchemes.Length)
+                {
+                    schemeRoundCounter = 0;
+                }
+
+                var scheme = locationSchemes[schemeRoundCounter];
+
+                return scheme;
+            });
+            var roller = rollerMock.Object;
+
+            var introScheme = locationSchemes.Single(x => x.Sid == "intro");
+
+            var biomService = new BiomeInitializer(sectorGenerator, roller);
+
+            var dice = new LinearDice();
+            var resourceMaterializationMap = new ResourceMaterializationMap(dice);
+
+            var resultStringBuilder = new StringBuilder();
+            resultStringBuilder.AppendLine("graph TD;");
+
+            // ACT
+
+            var biom = await biomService.InitBiomeAsync(introScheme).ConfigureAwait(false);
+            var introNode = biom.Sectors.Single(x => x.State == SectorNodeState.SectorMaterialized);
+            var currentResource = resourceMaterializationMap.GetDepositData(introNode);
+            var currentNode = introNode;
+
+            var iteration = 0;
+            const int ITERATION_MAX = 100;
+            while (iteration < ITERATION_MAX)
+            {
+                var nextNode = currentNode.Biome.GetNext(currentNode)
+                    .OfType<SectorNode>()
+                    .First(x => x.State != SectorNodeState.SectorMaterialized);
+
+                await biomService.MaterializeLevelAsync(nextNode).ConfigureAwait(false);
+                var nextResource = resourceMaterializationMap.GetDepositData(nextNode);
+
+                resultStringBuilder.AppendLine(GetVisualString(currentNode, nextNode, currentResource, nextResource));
+
+                currentNode = nextNode;
+                currentResource = nextResource;
+
+                iteration++;
+            }
+
+            var materializedSectorNodes = new List<SectorNode>();
+            var scanNode = introNode;
+            materializedSectorNodes.Add(scanNode);
+            while (true)
+            {
+                var nextNodes = scanNode.Biome.GetNext(scanNode).OfType<SectorNode>();
+                scanNode = nextNodes.SingleOrDefault(x => x.State == SectorNodeState.SectorMaterialized && !materializedSectorNodes.Contains(x));
+
+                if (scanNode is null)
+                {
+                    break;
+                }
+
+                materializedSectorNodes.Add(scanNode);
+            }
+
+            Console.Write(resultStringBuilder.ToString());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this logging statement.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5389027)